### ISSUE TITLE
Report compilation progress

### ISSUE THREAD

**Issue by aslushnikov** [aslushnikov/latex-online#29]
It would be nice to report progress for long-running compilation targets.

Examples of such a long-running compilations are:
- [introprog](https://latexonline.cc/compile?git=https%3A%2F%2Fgithub.com%2Flunduniversity%2Fintroprog&target=compendium%2Fcompendium.tex&command=pdflatex)
- [GOS_book](http://latex.aslushnikov.com/compile?git=https://github.com/DidenkoAndre/GOS_book&target=_main.tex)

**Comment by bjornregnell** [comment]
YES, this would be nice!

**Comment by bjornregnell** [comment]
Also, I was a bit confused by the message on the web page that said "compiling..." even if a cached version was opening in another window. I think that if there is already a pfd cached then the web page should say something like "Opening cached pdf..." so that the user does not think that the compilation is on-going when it's not.

**Comment by aslushnikov** [comment]
Right, the "compiling" statement is misleading. I expect it to be fixed simultaneously with implementation of progress reporting.

**Comment by MonsieurV** [comment]
Yep, it would be nice that the `/pending` endpoint reports the status of the current task. Maybe it could return a json payload that look like that when compiling:

```
{
  "id": "1495582367662",
  "compile": {
    "command": "pdflatex",
    "target": "resume.pdf",
    "type": "git",
    "url": "https://github.com/posquit0/Awesome-CV/blob/master/examples/resume.tex"
  },
  "status": "pending"
}
```

And when the task is over:

```
{
  "id": "1495582367662",
  "compile": {
    "command": "pdflatex",
    "target": "resume.pdf",
    "type": "git",
    "url": "https://github.com/posquit0/Awesome-CV/blob/master/examples/resume.tex"
  },
  "status": "done",
  "target_url": "https://latexonline.cc/target?id=1495582367662"
}
```

Well, something in the vein. Mainly, if we can get the status of the task and the url of the generated file when it's done, the interface would get maybe easier to use.

Also, I'm not sure the `/pending` method would be required anymore: if the `/compile` method return the json describing the task status, that would do it no?
